### PR TITLE
Fix abnormal page token cases in PHP sample gen

### PR DESCRIPTION
**Pull request by tcoffee-google**
Addresses #692.

### CODE REVIEW

**Comment by saicheems** [comment]
You'll also want to edit this line:
https://github.com/googleapis/toolkit/blob/66d2df4a4ce2166b4a4115b11686e4767c0e7665/src/main/java/com/google/api/codegen/discovery/transformer/php/PhpSampleMethodToViewTransformer.java#L102

So `optParams` won't appear if a page streaming method doesn't assign the page token through `optParams`.

**Comment by tcoffee-google** [comment]
> So `optParams` won't appear

Good catch.

File: src/main/java/com/google/api/codegen/discovery/transformer/php/PhpSampleMethodToViewTransformer.java [hunk 1]
```diff
@@ -169,7 +170,14 @@ private SamplePageStreamingView createSamplePageStreamingView(
     builder.isResourceMap(fieldInfo.type().isMap());
     builder.pageVarName(
         symbolTable.getNewSymbol(namer.localVarName(Name.lowerCamel(fieldInfo.name()))));
-    builder.pageTokenName(methodInfo.requestPageTokenName());
+    boolean isResourceSetterInRequestBody = methodInfo.isPageStreamingResourceSetterInRequestBody();
+    builder.isResourceSetterInRequestBody(isResourceSetterInRequestBody);
+    // Use upper camel case for getter/setter function names in templates.
+    if (isResourceSetterInRequestBody) {
+      builder.pageTokenName(Name.lowerCamel(methodInfo.requestPageTokenName()).toUpperCamel());
```
Comment (saicheems) [inline src/main/java/com/google/api/codegen/discovery/transformer/php/PhpSampleMethodToViewTransformer.java]: Use the namer for this, you can either add a `getPageTokenSetterFuncName` to the `PhpSampleNamer` or call `namer.publicMethodName(Name.lowerCamel("get", methodInfo.requestPageTokenName()))`

That way there's no name composition in the snippet

Comment (saicheems) [inline src/main/java/com/google/api/codegen/discovery/transformer/php/PhpSampleMethodToViewTransformer.java]: Oh, I see, that's how I did it in Java, huh? :/

If you could make an issue then so I can clean it up later that'll be enough.

Comment (tcoffee-google) [inline src/main/java/com/google/api/codegen/discovery/transformer/php/PhpSampleMethodToViewTransformer.java]: https://github.com/googleapis/toolkit/issues/991

File: src/main/resources/com/google/api/codegen/php/sample.snip [hunk 2]
```diff
@@ -125,6 +125,26 @@
   ?>
 @end
 
+@private nextPageToken(class)
+  ${@class.responseVarName}->get{@class.pageStreaming.nextPageTokenName}()
+@end
+
+@private updatePageToken(class)
+  @if class.pageStreaming.isResourceSetterInRequestBody
+    ${@class.requestBodyVarName}->set{@class.pageStreaming.pageTokenName}({@nextPageToken(class)});
```
Comment (saicheems) [inline src/main/resources/com/google/api/codegen/php/sample.snip]: Would also generally recommend not putting semicolons in the sub snippets and instead appending them to the actual call for consistency

```
...
updatePageToken(class);
```

**Comment by saicheems** [comment]
^ With that last comment, everything else seems fine, lgtm

**Pull request merged**